aptos-move/framework/aptos-framework/sources/governed_gas_pool.move

```diff
@@ -170,6 +172,27 @@ module aptos_framework::governed_gas_pool {
         coin::balance<CoinType>(pool_address)
     }
 
+    /// Withdraws coins from the governed gas pool.
+    ///
+    /// This function allows friend modules to withdraw a specified amount of a given
+    /// `CoinType` from the governed gas pool. It uses the internal signer of the
+    /// governed gas pool to authorize the withdrawal.
+    ///
+    /// @param amount The amount of coins to withdraw from the pool.
+    /// @return A `Coin<CoinType>` resource containing the withdrawn amount.
+    public(friend) fun withdraw_from_pool<CoinType>(
```

Comment:
have a dedicated function called withdraw_staking_reward and add dedciated event for it